File: backend/internal/domain/mock.go
package domain

import (
	"time"
)

type MockAPI struct {
	ID           string    `json:"id"`
	UserID       string    `json:"user_id"`
	Path         string    `json:"path"`
	Method       string    `json:"method"`
	Status       int       `json:"status"`
	ResponseBody string    `json:"response_body"`
	CreatedAt    time.Time `json:"created_at"`
	ExpiresAt    time.Time `json:"expires_at"`
	HitCount     int       `json:"hit_count"`
}

File: frontend/src/types/mock.ts
export type MockEndpoint = {
    id: string;
    user_id: string;
    path: string;
    method: string;
    status: number;
    response_body: string;
    created_at: string;
    expires_at: string;
    hit_count?: number;
    curl_command?: string;
};
